File: backend/app/schemas/homepage.py
"""
Homepage content schemas for API requests/responses.
"""
from datetime import datetime
from typing import Optional, Any, List
from pydantic import BaseModel


class HomepageContentBase(BaseModel):
    key: str
    content: dict
    is_active: Optional[bool] = True


class HomepageContentCreate(HomepageContentBase):
    pass


class HomepageContentUpdate(BaseModel):
    content: Optional[dict] = None
    is_active: Optional[bool] = None


class HomepageContentResponse(HomepageContentBase):
    id: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


# Hero section content structure
class HeroContent(BaseModel):
    headline: str
    tagline: str
    subtitle: str


# Interest item structure
class InterestItem(BaseModel):
    icon: str
    title: str
    description: str


class InterestsContent(BaseModel):
    interests: List[InterestItem]

File: frontend/src/services/homepage.ts
import api from './api'

// Types
export interface HomepageContent {
  id: string
  key: string
  content: Record<string, unknown>
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface HeroContent {
  headline: string
  tagline: string
  subtitle: string
}

export interface InterestItem {
  icon: string
  title: string
  description: string
}

export interface InterestsContent {
  interests: InterestItem[]
}

// API functions
export async function getHomepageContent(): Promise<HomepageContent[]> {
  const response = await api.get<HomepageContent[]>('/homepage/content')
  return response.data
}

export async function getHomepageContentByKey(key: string): Promise<HomepageContent> {
  const response = await api.get<HomepageContent>(`/homepage/content/${key}`)
  return response.data
}

export async function getHeroContent(): Promise<HeroContent> {
  const response = await api.get<HomepageContent>('/homepage/content/hero')
  return response.data.content as unknown as HeroContent
}

export async function getInterestsContent(): Promise<InterestsContent> {
  const response = await api.get<HomepageContent>('/homepage/content/interests')
  return response.data.content as unknown as InterestsContent
}

// Admin functions
export async function updateHomepageContent(
  key: string,
  content: Record<string, unknown>
): Promise<HomepageContent> {
  const response = await api.put<HomepageContent>(`/homepage/admin/content/${key}`, {
    content,
  })
  return response.data
}

export async function createHomepageContent(
  key: string,
  content: Record<string, unknown>
): Promise<HomepageContent> {
  const response = await api.post<HomepageContent>('/homepage/admin/content', {
    key,
    content,
  })
  return response.data
}
